tasks/component-usage.js

```diff
@@ -0,0 +1,163 @@
+    short: 'd',
+  },
+  help: {
+    type: 'boolean',
+    short: 'h',
+  },
+  'used-by': {
+    type: 'string',
+  },
+  uses: {
+    type: 'string',
+    short: 'u',
+  },
+};
+
+const { values } = parseArgs({ args, options });
+
+const depth = (values?.depth) ? Number(values.depth) : null;
+const all = (values?.all) ? Infinity : null;
+const uses = values?.uses ? String(values?.uses) : null;
+const usedBy = values?.['used-by'] ? String(values?.['used-by']) : null;
+
+if (values?.help) {
+  console.log(`
+    Command: node tasks/component-usage.js --depth number|--all --uses|--used-by global|cc-component
+    --depth (alias -d): Depth of the tree - value: number. 
+    --all (alias -a): The depth of tree will be infinity - value: no value.
+    --uses (alias -u): Lists the components that use the component given in value param - value: global for all the components|cc-component for a specific component. 
+    --used-by : Lists the components needed by the component given in value param - value: global for all the components|cc-component for a specific component.
+    
+    Note: 
+    * --uses or --used-by must be provided.
+    * --uses and --used-by can't be used together
+    * --depth and --all can't be used together.
+    * The depth of the tree will be:
+      * 1 for global by default
+      * Infinity when specifying a component by default 
+  `);
+  process.exit(0);
+}
+
+if (isNaN(depth)) {
+  console.log(
+    chalk
+      .red
+      .bold(`❌ Depth provided was not a number.`),
+  );
+  process.exit(1);
+}
+
+if (all != null && depth != null) {
+  console.log(
+    chalk
+      .red
+      .bold(`❌ Can't have both --depth and --all together.`),
+  );
+  process.exit(1);
+}
+
+if (uses != null && usedBy != null) {
+  console.log(
+    chalk
+      .red
+      .bold(`❌ Can't have --uses and --used-by together.`),
+  );
+  process.exit(1);
+}
+else if (uses == null && usedBy == null) {
+  console.log(
+    chalk
+      .red
+      .bold(`❌ --uses or --used-by param is missing. Use --help for more information.`),
+  );
+  process.exit(1);
+
+}
+
+if ((uses != null && !isComponent(uses)) || (usedBy != null && !isComponent(usedBy))) {
+  console.log(
+    chalk
+      .red
+      .bold(`❌ The argument you provided doesn't seem to be a component.`),
+  );
+  process.exit(1);
+}
+
+let component = '';
+let mode = null;
+let maxLevel = 1;
+
+if (uses != null && uses === 'global') {
+  mode = 'uses-g';
+  maxLevel = Infinity;
+  component = 'global';
+}
+else if (uses != null) {
+  mode = 'uses';
+  component = uses;
+}
+else if (usedBy != null && usedBy === 'global') {
+  mode = 'used-by';
+  component = 'global';
+  maxLevel = Infinity;
+}
+else if (usedBy != null) {
+  mode = 'used-by';
+  component = usedBy;
+}
+
+if (depth != null) {
+  maxLevel = depth;
+}
+else if (all != null) {
+  maxLevel = all;
+}
+
+function isComponent (input) {
+  return input.startsWith('cc-') || input === 'global';
+}
+
+let componentsInput = '';
```

Comment:
I'm not very confortable with this variable which can be either a string or an array of string. And, then is passed to function that can not handle both types (`getDependantsLsStyle` only handle string while `getComponentsTree` only handle array).

Reply:
I was thinking of making it an array every time and for the 'uses' mode I'll pass `componentsInput[0]` to the `getDependantsLsStyle` function. WDYT?

Reply:
ok